### PR TITLE
Change DLC to NNPA for consistency in accelerator name

### PR DESCRIPTION
**Pull request by tungld**
This patch replaces  DLC by NNPA for consistency in accelerator name.

### CODE REVIEW

File: src/Accelerators/NNPA/CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # SPDX-License-Identifier: Apache-2.0
 
-set(DLC_SRC_ROOT "${CMAKE_CURRENT_SOURCE_DIR}")
-set(DLC_BIN_ROOT "${CMAKE_CURRENT_BINARY_DIR}")
+set(NNPA_SRC_ROOT "${CMAKE_CURRENT_SOURCE_DIR}")
```
Comment (AlexandreEichenberger) [inline src/Accelerators/NNPA/CMakeLists.txt:3]: Do we need the repliacated names? e.g. 
```
set(NNPA_ONNX_MLIR_SRC_ROOT ${ONNX_MLIR_SRC_ROOT})
 set(NNPA_ONNX_MLIR_BIN_ROOT ${ONNX_MLIR_BIN_ROOT})
```
I don't think we need to have an `NNPA_ONNX_MLIR_SRC_ROOT` to begin with. 
Let me know  if you prefer having this in a subsequent PR

File: src/Accelerators/NNPA/Conversion/ZLowToLLVM/CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,6 @@ add_onnx_mlir_library(OMZLowToLLVM
   OMZLowOps
 
   ACCEL_INCLUDE_DIRS PRIVATE
-  ${DLC_SRC_ROOT}
-  ${DLC_BIN_ROOT}
+  ${NNPA_SRC_ROOT}
```
Comment (AlexandreEichenberger) [inline src/Accelerators/NNPA/Conversion/ZLowToLLVM/CMakeLists.txt:12]: This should be relative to onnx-mlir source dir

File: src/Accelerators/NNPA/Conversion/ONNXToZHigh/NNPALimit.h [hunk 3]
```diff
@@ -16,12 +16,12 @@
 
 // The NNPA maximum supported dimension index size value by using
 // zdnn_get_nnpa_max_dim_idx_size() This value depends on HW.
-static constexpr int64_t DLCPP_MAXIMUM_DIMENSION_INDEX_SIZE = 32768;
+static constexpr int64_t NNPAPP_MAXIMUM_DIMENSION_INDEX_SIZE = 32768;
```
Comment (etiotto) [inline src/Accelerators/NNPA/Conversion/ONNXToZHigh/NNPALimit.h:19]: NNPAPP_MAXIMUM_DIMENSION_INDEX_SIZE -> NNPA_MAXIMUM_DIMENSION_INDEX_SIZE

File: src/Accelerators/NNPA/Conversion/ONNXToZHigh/NNPALimit.h [hunk 4]
```diff
@@ -16,12 +16,12 @@
...
 
 // The NNPA maximum supported tensor size (in bytes)
 // by using zdnn_get_nnpa_max_tensor_size()
 // This value depends on HW.
-static constexpr int64_t DLCPP_MAXIMUM_TENSOR_SIZE = 4294967296;
+static constexpr int64_t NNPAPP_MAXIMUM_TENSOR_SIZE = 4294967296;
```
Comment (etiotto) [inline src/Accelerators/NNPA/Conversion/ONNXToZHigh/NNPALimit.h:24]: NNPAPP_MAXIMUM_TENSOR_SIZE -> NNPA_MAXIMUM_TENSOR_SIZE

File: src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXToZHighCommon.hpp [hunk 5]
```diff
@@ -33,15 +33,15 @@ void addDynamicallyLegalOpFor(mlir::ConversionTarget *target,
     }
 
     // Check zDNN limitations
-    // TODO: Check tensor size DLCPP_MAXIMUM_TENSOR_SIZE of another limitation
+    // TODO: Check tensor size NNPAPP_MAXIMUM_TENSOR_SIZE of another limitation
```
Comment (etiotto) [inline src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXToZHighCommon.hpp:36]: NNPAPP_MAXIMUM_TENSOR_SIZE -> NNPA_MAXIMUM_TENSOR_SIZE

File: src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXToZHighCommon.hpp [hunk 6]
```diff
@@ -33,15 +33,15 @@ void addDynamicallyLegalOpFor(mlir::ConversionTarget *target,
...
     bool exceedLimit =
         llvm::any_of(genericOp->getOperands(), [](Value operand) {
           if (auto valueType = operand.getType().dyn_cast<ShapedType>()) {
             // Check if static dimension size exceeds zDNN limitations
             ArrayRef<int64_t> valueShape = valueType.getShape();
             if (llvm::any_of(valueShape, [](int64_t dim) {
                   return (dim != -1) &&
-                         (dim > DLCPP_MAXIMUM_DIMENSION_INDEX_SIZE);
+                         (dim > NNPAPP_MAXIMUM_DIMENSION_INDEX_SIZE);
```
Comment (etiotto) [inline src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXToZHighCommon.hpp:44]: NNPAPP_MAXIMUM_DIMENSION_INDEX_SIZE -> NNPA_MAXIMUM_DIMENSION_INDEX_SIZE

File: src/Accelerators/NNPA/CMakeLists.txt (hunk 1 above)
Comment (tungld) [reply src/Accelerators/NNPA/CMakeLists.txt:3]: Yes, leave it for another PR to keep this patch self-contained.

File: src/Accelerators/NNPA/Conversion/ONNXToZHigh/NNPALimit.h (hunk 3 above)
Comment (tungld) [reply src/Accelerators/NNPA/Conversion/ONNXToZHigh/NNPALimit.h:19]: Good catch! Thanks!

File: src/Accelerators/NNPA/Conversion/ZLowToLLVM/CMakeLists.txt (hunk 2 above)
Comment (tungld) [reply src/Accelerators/NNPA/Conversion/ZLowToLLVM/CMakeLists.txt]: Done. This can be removed.

**Review by etiotto — APPROVED** [review]
LGTM

**Pull request merged**